File: 0263-ugly-number/0263-ugly-number.cpp
class Solution {
public:
    int dividing(int dividend, int divisor){
        while(dividend%divisor==0){
            dividend = (int)dividend/divisor;
        }
        return dividend;
    }
    bool isUgly(int n) {
        if(n<=0){
            return false;
        }
        for(auto factor: {2,3,5}){
            n = dividing(n, factor);
        }
        
        return n==1;
        
    }
};

File: 0263-ugly-number/0263-ugly-number.py
class Solution:
    def isUgly(self, n: int) -> bool:
        if n<=0:
            return False
        
        def keepDividingBy(dividend, divisor):
            while dividend%divisor == 0:
                dividend //= divisor
            return dividend
        
        for factor in [2,3,5]:
            n = keepDividingBy(n, factor)
        
        return n == 1